subsys/net/lib/http/http_client.c

```diff
@@ -411,17 +383,25 @@ static void http_client_init_parser(struct http_parser *parser,
 static int http_wait_data(int sock, struct http_request *req)
 {
 	int total_received = 0;
-	size_t offset = 0;
 	int received, ret;
 
 	do {
-		received = zsock_recv(sock, req->internal.response.recv_buf + offset,
-				req->internal.response.recv_buf_len - offset,
-				0);
+		received = zsock_recv(sock, req->internal.response.recv_buf,
```

Comment:
What happens if we receive data one byte at a time, then the earlier data is discarded after each recv call

Reply:
The parser processes the data one byte at a time (https://github.com/zephyrproject-rtos/zephyr/blob/50ff77fa910dcad29167b389a6c7c67ade8f6f17/subsys/net/lib/http/http_parser.c#L784) so it's shouldn't be an issue for the parser.

If we want to restore the "offsetting" functionality to report larger chunks to the application in such case, I can do this, it'd complicate the logic a bit however.

Reply:
Actually, it wasn't that bad - I've restored the offset and updated the logic - less noise on the application side with chunked responses.

Reply:
:+1: thanks! I think it is better if application can receive more than one byte at a time.